geoh5py/objects/text.py

```diff
@@ -0,0 +1,225 @@
+    )
+    layer_name: str | None = Field(
+        validation_alias=AliasChoices("layer_name", "LayerName"),
+        serialization_alias="LayerName",
+        default="",
+    )
+    normal: str | None = Field(
+        validation_alias=AliasChoices("normal", "Normal"),
+        serialization_alias="Normal",
+        default="{0,0,1}",
+    )
+    obliquing_angle: int | None = Field(
+        validation_alias=AliasChoices("obliquing_angle", "ObliquingAngle"),
+        serialization_alias="ObliquingAngle",
+        default=0,
+    )
+    rotation_angle: int | None = Field(
+        validation_alias=AliasChoices("rotation_angle", "RotationAngle"),
+        serialization_alias="RotationAngle",
+        default=0,
+    )
+    text: str | None = Field(
+        validation_alias=AliasChoices("text", "Text"),
+        serialization_alias="Text",
+        default="<no-data>",
+    )
+    width_scale_factor: float | None = Field(
+        validation_alias=AliasChoices("width_scale_factor", "WidthScaleFactor"),
+        serialization_alias="WidthScaleFactor",
+        default=1.0,
+    )
+
+    @field_validator("starting_point", "normal", "direction", mode="before")
+    @classmethod
+    def array_to_str(cls, float_list: np.ndarray | list):
+        if isinstance(float_list, np.ndarray | list):
+            float_list = "{" + ",".join([str(val) for val in float_list]) + "}"
+
+        return float_list
+
+
+class TextData(BaseModel):
+    """
+    Text mesh data.
+    """
+
+    text_data: list[TextEntry] = Field(
+        validation_alias=AliasChoices("text_data", "Text Data"),
+        serialization_alias="Text Data",
+    )
+
+
+class TextObject(Points):
+    """
+    Text object for annotated text labels in viewport.
+    """
+
+    _attribute_map: dict = Points._attribute_map.copy()
+    _attribute_map.update({"TextMesh Data": "text_mesh_data"})
+    _default_name = "Text Object"
+    _TYPE_UID: uuid.UUID | None = uuid.UUID("{c00905d1-bc3b-4d12-9f93-07fcf1450270}")
+
+    def __init__(
+        self,
+        vertices: np.ndarray | list | tuple | None = None,
+        text_mesh_data: str | None = None,
+        **kwargs,
+    ):
+        self._text_mesh_data: TextData
+
+        super().__init__(vertices=vertices, text_mesh_data=text_mesh_data, **kwargs)
+
+    @property
+    def text_mesh_data(self) -> TextData:
+        """
+        Text mesh data.
+        """
+
+        return self._text_mesh_data
+
+    @text_mesh_data.setter
+    def text_mesh_data(self, value: str | None):
+        """
+        Set the text mesh data.
+        """
+
+        self._text_mesh_data = self._validate_text_data(value)
+
+    @property
+    def extent(self):
+        """
+        Geography bounding box of the object.
+        """
+        return None
+
+    def mask_by_extent(self, extent: ndarray, inverse: bool = False) -> None:
+        """
+        Sub-class extension of :func:`~geoh5py.shared.entity.Entity.mask_by_extent`.
+        """
+        return None
```

Comment:
Should this just return an array of all true?

Reply:
I guess same comment goes for the base class behaviour:
![image](https://github.com/user-attachments/assets/4dc8255e-2779-4dfb-902f-b05879f6a6c3)

Reply:
Shouldn't have to check if the result is None everywhere before we use the mask.

Reply:
The goal of `mask_by_extent` was to have a generic method for all entities, even those that don't have coordinates (like labels and groups). If a group does have objects, then its `mask_by_extent` will return the combine extent of everything, otherwise it won't be copied. 

By setting to None here I am saying that this object can't be copied by extent (until we write the mechanics to update the locations in the text_data at the same time as vertices).